src/components/CloudBrute/CloudBrute.tsx

```diff
@@ -0,0 +1,173 @@
+            })
+            .catch((error) => {
+                console.error("An error occurred:", error);
+                setLoadingModal(false); // Also set loading to false in case of error
+            });
+    }, []);
+
+    /**
+     * handleProcessData: Callback to handle and append new data from the child process to the output.
+     * It updates the state by appending the new data received to the existing output.
+     * @param {string} data - The data received from the child process.
+     */
+    const handleProcessData = useCallback((data: string) => {
+        setOutput((prevOutput) => prevOutput + "\n" + data); // Append new data to the previous output.
+    }, []);
+
+    /**
+     * handleProcessTermination: Callback to handle the termination of the child process.
+     * Once the process termination is handled, it clears the process PID reference and
+     * deactivates the loading overlay.
+     * @param {object} param - An object containing information about the process termination.
+     * @param {number} param.code - The exit code of the terminated process.
+     * @param {number} param.signal - The signal code indicating how the process was terminated.
+     */
+    const handleProcessTermination = useCallback(
+        ({ code, signal }: { code: number; signal: number }) => {
+            if (code === 0) {
+                handleProcessData("\nProcess completed successfully.");
+            } else if (signal === 15) {
+                handleProcessData("\nProcess was manually terminated.");
+            } else {
+                handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+
+            setPid(""); // Clear the child process pid reference.
+            setLoading(false); // Cancel the loading overlay.
+        },
+        [handleProcessData] // Dependency on the handleProcessData callback
+    );
+
+    /**
+     * onSubmit: Asynchronous handler for the form submission event.
+     * It sets up and triggers the CloudBrute tool with the given parameters.
+     * Once the command is executed, the results or errors are displayed in the output.
+     *
+     * @param {FormValuesType} values - The form values, containing the domain, keyword, and wordlist path.
+     */
+    const onSubmit = async (values: FormValuesType) => {
+        setLoading(true);
+        const args = ["-d", values.domain, "-k", values.keyword, "-w", values.wordlist];
+        CommandHelper.runCommandGetPidAndOutput("cloudbrute", args, handleProcessData, handleProcessTermination)
+            .then(({ pid, output }) => {
+                setPid(pid);
+                setOutput(output);
+            })
+            .catch((error) => {
+                setLoading(false);
+                setOutput(`Error: ${error.message}`);
+            });
+    };
+
+    /**
+     * Clears the output state.
+     */
+    const clearOutput = useCallback(() => {
+        setOutput("");
+    }, [setOutput]);
```

Comment:
You missed  setAllowSave, setHasSaved and also the Callback function to handle the completion of a save operation.

```suggestion
    const clearOutput = useCallback(() => {
        setOutput("");
        setAllowSave(false);
        setHasSaved(false);
    }, [setOutput]);

     const handleSaveComplete = useCallback(() => {
        // Indicating that the file has saved which is passed
        // back into SaveOutputToTextFile to inform the user
        setHasSaved(true);
        setAllowSave(false);
    }, []);
```